server/src/test/java/org/opensearch/index/shard/RemoteStoreRefreshListenerTests.java

```diff
@@ -52,7 +53,7 @@ public void setup(boolean primary, int numberOfDocs) throws IOException {
         indexDocs(1, numberOfDocs);
         indexShard.refresh("test");
 
-        remoteStoreRefreshListener = new RemoteStoreRefreshListener(indexShard);
+        remoteStoreRefreshListener = new RemoteStoreRefreshListener(indexShard, SegmentReplicationCheckpointPublisher.EMPTY);
```

Comment:
Can we please also add tests by passing maybe a mock of `SegmentReplicationCheckpointPublisher` and verifying that it is getting invoked as expected?